pkg/provision/engine.go

```diff
@@ -95,11 +99,17 @@ func New(opts EngineOps) *Engine {
 func (e *Engine) Run(ctx context.Context) error {
 	cReservation := e.source.Reservations(ctx)
 
+	memStats, err := mem.VirtualMemory()
+	if err != nil {
+		log.Fatal().Err(err).Msg("failed retrieve memory stats")
```

Comment:
please return the error, don't do log.Fatal here. it's not up to this method to decide.
Also this should have been done in the NewMethod, not the run.